File: src/plex/apis/file.js
class FileClient {
    constructor(client) {
        this._client = client;
    }

    async getFileBuffer(file, startIndex, numberOfBytes) {
        if (startIndex >= file.size) {
            // nothing to do, we have read 0 bytes
            return null;
        }

        const endIndex = Math.min(startIndex + numberOfBytes - 1, file.size - 1);

        return await this._client.bufferFetch(file.server.url, file.next, {
            'X-Plex-Token': file.server.token,
            'download': 1
        }, startIndex, endIndex);
    }
}

module.exports = FileClient;

File: plex/apis/File.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Pfs.Plex.Apis
{
    public class FileClient
    {
        private BaseClient _client;
        public FileClient(BaseClient client)
        {
            this._client = client;
        }

        // public async Task<byte[]> GetFileBuffer(Node file, long startIndex, long numberOfBytes)
        // {
        //     if (startIndex >= file.Size) {
        //         // nothing to do, we have read 0 bytes
        //         return null;
        //     }

        //     var endIndex = Math.Min(startIndex + numberOfBytes - 1, file.Size - 1);

        //     return await this._client.BufferFetch(file.Server.Url, file.Next, {
        //         { "X-Plex-Token", file.Server.Token },
        //         { "download", "1" }
        //     }, startIndex, endIndex);
        // }
    }
}
